pkg/kubelet/kubelet_test.go

```diff
@@ -486,6 +485,54 @@ func TestHandlePortConflicts(t *testing.T) {
 	checkPodStatus(t, kl, fittingPod, v1.PodPending)
 }
 
+// Tests serial pod admit that we handle port conflicts correctly by setting the failed status in status map.
+func TestHandleSerialPodAdmit(t *testing.T) {
+	testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
+	defer testKubelet.Cleanup()
+	kl := testKubelet.kubelet
+
+	kl.nodeInfo = testNodeInfo{nodes: []*v1.Node{
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: string(kl.nodeName)},
+			Status: v1.NodeStatus{
+				Allocatable: v1.ResourceList{
+					v1.ResourcePods: *resource.NewQuantity(110, resource.DecimalSI),
+				},
+			},
+		},
+	}}
+	recorder := record.NewFakeRecorder(20)
+	nodeRef := &v1.ObjectReference{
+		Kind:      "Node",
+		Name:      string("testNode"),
+		UID:       types.UID("testNode"),
+		Namespace: "",
+	}
+	testClusterDNSDomain := "TEST"
+	kl.dnsConfigurer = dns.NewConfigurer(recorder, nodeRef, nil, nil, testClusterDNSDomain, "")
+	spec := v1.PodSpec{NodeName: string(kl.nodeName), Containers: []v1.Container{{Ports: []v1.ContainerPort{{HostPort: 80}}}}}
+	notfittingPod := []*v1.Pod{
+		podWithUIDNameNsSpec("123456789", "newpod", "foo", spec),
+	}
+	fittingPod := []*v1.Pod{
+		podWithUIDNameNsSpec("987654321", "oldpod", "foo", spec),
+	}
+
+	// Make sure the Pods are in the reverse order of creation time.
+	notfittingPod[0].CreationTimestamp = metav1.NewTime(time.Now().Add(2 * time.Second))
+	fittingPod[0].CreationTimestamp = metav1.NewTime(time.Now().Add(1 * time.Second))
+
+	kl.HandlePodAdditions(notfittingPod)
```

Comment:
the sorting code is exercised within a single invocation of HandlePodAdditions... making two calls with a single pod each does not exercise sorting, right?

I would have expected a single invocation of HandlePodAdditions with multiple conflicting pods provided out of order, and a test verifying that the correct pods are marked as pending and failed based on the expected sort order